openassessment/assessment/models/staff.py

```diff
@@ -133,6 +134,13 @@ def get_submission_for_review(cls, course_id, item_id, scorer_id):
                     grading_completed_at=None,
                     cancelled_at=None,
                 )
+                # if filter_by_teams, keep only staff_workflows from students in the same teams as staff
+                # this is just pseudo code (relations have not been investigated)
+                if filter_by_teams:
+                    staff_teams = Teams.objects.filter(user_id=scorer_id)
+                    staff_workflows = staff_workflows.filter(
+                        submission__student__teams__intersect_with=staff_teams
+                    )
```

Comment:
Tracking the function render_staff_grade_form, this is where it all ends up, this is where we would need to implement the teams filtering.